### PR TITLE
I1297 create big tree project

### PR DESCRIPTION
**Pull request by lidaqing**
Can create a big tree project now

### CODE REVIEW

File: tools/smv-init [hunk 1]
```diff
@@ -11,7 +11,7 @@ OVERRIDE=0
 DO_GIT_INIT=0
 
 
-while [[ $# -gt 1 ]]
+while [[ $# -gt 2 ]]
```
Comment (cloudysail) [inline tools/smv-init]: This should not change. `PROJ_SIZE` needs to be an optional parameter for `bc` and `bt`
Need to make sure old function still works:
```
➜  SMV git:(i1297_create_big_tree_project) ./tools/smv-init ../test
ERROR: Invalid number of arguments
USAGE: ./tools/smv-init [-q][-o] [-s|-e|-t|-b|-bc|-bt] [-dd] [-n]  project_path
project types:
  -s: simple (default)
  -e: enterprise
  -t: test (for developers only)
  -b: blank/empty project
  -bc: big chain project
  -bt: big tree project
  -dd: data directory
  -n: project name
  -g: initializes git repo in project directory

example:
    simple:     $ ./tools/smv-init /project/path
    blank:      $ ./tools/smv-init -b -n projec_name -dd /data/dir /project/path
    enterprise: $ ./tools/smv-init -e /project/path
```

File: tools/templates/bigTree/dupMod2.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
```
Comment (cloudysail) [inline tools/templates/bigTree/dupMod2.sh]: `dupMod2.sh` is not a good file name, and there are no comments for it!
Need to think about how to make other developers easily understand it. People will ask what is dupMod.sh and dupMod2.sh doing? What's the difference?

**Comment by lidaqing** [comment]
Rename file name and the parameter back to 1

File: tools/templates/bigTree/dupTreeMod.sh [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+sed -i "" "s/_DEP_NAME_/input/" "$DEST/$DUPNAME"
+
+for((i=1;i<$NUM_MODS;i++)); do
+LASTDUPNAME=$DUPNAME
+DUPNAME="$(echo $TBNAME | sed "s/\./$i./")"
+cp "$TEMPLATE" "$DEST/$DUPNAME"
+sed -i "" "s/_MOD_NAME_/M$i/" "$DEST/$DUPNAME"
+VAR=i%2
+if [ $VAR = 1 ]; then
+  sed -i "" "s/_PKG_NAME_/stage1.${TBNAME%.py}$(i/2)/" "$DEST/$DUPNAME"
```
Comment (laneb) [inline tools/templates/bigTree/dupTreeMod.sh]: I think `sed -i` may not work on linux...I'm pretty sure we've run into issues with it before. Does the use case for require need usability on linux?

File: tools/smv-init [hunk 4]
```diff
@@ -68,14 +77,17 @@ TEMPLATE_DATA_DIR="$ALL_TEMPLATES_DIR/data"
...
     echo "ERROR: Invalid number of arguments"
-    echo "USAGE: $0 [-q][-o] [-s|-e|-t|-b] [-dd] [-n]  project_path"
+    echo "USAGE: $0 [-q][-o] [-s|-e|-t|-b|-bc|-bt] [-dd] [-n] [-mn] project_path"
     echo "project types:"
     echo "  -s: simple (default)"
     echo "  -e: enterprise"
     echo "  -t: test (for developers only)"
     echo "  -b: blank/empty project"
-    echo "  -d: data directory"
+    echo "  -bc: big chain project"
```
Comment (laneb) [inline tools/smv-init]: Short options should have a single letter.

File: tools/templates/bigTree/src/main/python/stage1/employment.py [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import smv
+import pyspark.sql.functions as F
+
+__all__ = ['EmploymentByState']
```
Comment (laneb) [inline tools/templates/bigTree/src/main/python/stage1/employment.py]: Remove this for clarity.

File: tools/templates/bigTree/dupTreeMod.sh (hunk 3 above)
Comment (AliTajeldin) [reply tools/templates/bigTree/dupTreeMod.sh]: yup.  We usually revert to `perl -pi` instead.

File: tools/smv-init [hunk 6]
```diff
@@ -50,6 +54,11 @@ FLAG=$1
         shift
         shift
         ;;
+    -mn|--moduleNumber)
```
Comment (cloudysail) [inline tools/smv-init]: still, there are some -xx

https://serverfault.com/questions/387935/whats-the-difference-betwen-the-single-dash-and-double-dash-flags-on-shell-comm

**Comment by lidaqing** [comment]
Have updated

**Pull request merged**